### ISSUE TITLE
Rework mariadb facts

### ISSUE THREAD

**Issue by ganto** [debops/ansible-mariadb_server#37]
@drybjed in debops/debops-policy#16  you wrote:

> The facts in the debops.mariadb and debops.mariadb_server roles were written some time ago. At present time, I would write them as a Python script and probably make them more dynamic, for example to check if values defined in the inventory/defaults need to be updated. In the past I used facts as static information set in stone, but that is changing recently.

Can you maybe explain a bit more, what you mean by this? What would such a python script do?

**Comment by drybjed** [comment]
With the MariaDB roles, first of all I would check if the database server is installed on the host - that would change how the client role behaves. Perhaps the facts could parse `/etc/mysql/` configuration files and get the name of the database host from there. Of course some facts could still be defined by Ansible and stored in the fact script itself.

Maybe the script could check if an `stunnel` process is present, parse its configuration file and determine if the MariaDB server connection is tunneled over it; although with TLS connections to the database now working fine that wouldn't be as useful anymore.

Basically, you don't need to have the fact be a script; it can be static. Buf it there is some information that can be gathered dynamically, it might be better to get that information via a script instead of keeping it statically in the static fact file, since this information could have been changed in the meantine. Getting accurate facts about hosts is always better than using stale data.

Do you have any ideas about the MariaDB facts you want to implement?

**Comment by ganto** [comment]
Ok, I see. Thanks for the clarification. 

> Do you have any ideas about the MariaDB facts you want to implement?

I still have this idea about the cluster bootstrapping :smirk:  For this I would like to set some cluster details (bind address, port, cluster name, ...) as facts. So that I could generate the local cluster address.

_*thinking loud*..._
Is there anything that speaks against installing such a fact script at the beginning? Then then the final configuration could be considered before writing the own. Although having the facts dynamical wouldn't work in this case as the configuration files are not written then. On the other side it would simplify the cluster restart when using the [Safe-to-Bootstrap](http://galeracluster.com/2016/11/introducing-the-safe-to-bootstrap-feature-in-galera-cluster/) feature where the flag could be returned as fact.

Need to think more...

**Comment by drybjed** [comment]
Sure, setting up facts at the start of the role and reading them can be done.

**Comment by drybjed** [comment]
I'm closing this issue for now so that it doesn't block the merge of DebOps roles into unified `git` repository. It can be revisited later, epseically when both MariaDB-related roles will be in one repository which will make synchronization of facts easier.
